src/java/org/apache/cassandra/guardrails/Guardrail.java

```diff
@@ -59,64 +59,55 @@
 {
     private static final NoSpamLogger logger = NoSpamLogger.getLogger(LoggerFactory.getLogger(Guardrail.class),
```

Comment:
Hmm, yeah I don't know the history or reason about that; it looks like the `NoSpamLogger` has been there since the start and the `skipNotifying` mechanism was added in DB-3209.

Maybe the `skipNotifying` was added mostly for limiting `Tracing` and `Guardrails.Listener` and the potential effect on log messages was overlooked?

Since this is the way it is currently working in cndb now, I'm inclined to leave it for now, but if you think otherwise I can also remove the `NoSpamLogger`.

Reply:
@JeremiahDJordan what do you think? Do you know why ^ is implemented that way?

Reply:
I think we want to use NoSpamLogger.  The point here is to prevent spamming log files when the messages are coming from users queries.

Reply:
It was decided that `NoSpamLogger` should remain